src/event.js

```diff
@@ -36,24 +11,22 @@ const emitEvent = (kinesis, event, config) => {
   return kinesis.putRecord(params).promise();
 };
 
-const emitEvents = (kinesis, events, config) => {
-  const records = events.map(event => {
-    const enrichedEvent = enrichMeta(event, config.appName, config.ipv4);
-    return {
-      Data: JSON.stringify(enrichedEvent),
-      PartitionKey: config.partitionKey || partitionKey(enrichedEvent),
-    }
-  })
-
-  const params = {
-    Records: records,
-    StreamName: config.kinesisStream.resource
-  };
-
-  return kinesis.putRecords(params).promise();
+const emitEventWithRetry = async (kinesis, event, config, retries) => {
+  try {
+      return await emitEvent(kinesis, event, config)
+  } catch(error) {
+      if (retries === 0) throw error;
+      return await emitEventWithRetry(kinesis, event, config, retries - 1);
+  }
 };
 
-module.exports = {
-  emitEvent,
-  emitEvents,
+module.exports = (kinesis, event, config) => {
+  const retries = config.maxRetries || 0;
+  return new Promise((resolve, reject) => {
```

Comment:
I think here you should be able to just do `return emitEventWithRetry(kinesis, event, config, retries)` no?

Reply:
And I think this function should also be async as it's trying to `try`/`catch` the call to an async function.
And if keeping the promise style, you would still need `resolve(await emitEventWithRetry())` I believe